src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1371,13 +1460,25 @@ private bool ShouldAsyncBeTreatedAsModifier(bool parsingStatementNotDeclaration)
 
         private static bool IsNonContextualModifier(SyntaxToken nextToken)
         {
-            return GetModifier(nextToken) != DeclarationModifiers.None && !SyntaxFacts.IsContextualKeyword(nextToken.ContextualKind);
+            return GetModifier(nextToken.Kind, contextualKind: SyntaxKind.None) != DeclarationModifiers.None;
         }
 
-        private bool IsPartialType()
+        private bool IsTypeDeclaration(out SyntaxToken lastToken)
```

Comment:
'lastToken' is super vague here.  Why do you need this to be passed out.  Does some caller actually have to check something about htat, as opposed to just reading in the modifiers, and then figuring out what to do when they get to their own token that is DeclaraitonModifiers.None?

Reply:
we have something like this elsewhere, see `ScanType(out lastTokenOfType)`